starter-code/app.js

```diff
@@ -1,58 +1,90 @@
+const cookieParser = require("cookie-parser");
+const express = require("express");
+const favicon = require("serve-favicon");
+const hbs = require("hbs");
+const mongoose = require("mongoose");
+const logger = require("morgan");
+const path = require("path");
+const session = require("express-session");
+const MongoStore = require("connect-mongo")(session);
 
 mongoose
-  .connect('mongodb://localhost/starter-code', {useNewUrlParser: true})
+  .connect(process.env.DBURL, {
+    useNewUrlParser: true,
+    useUnifiedTopology: true
+  })
   .then(x => {
-    console.log(`Connected to Mongo! Database name: "${x.connections[0].name}"`)
+    console.log(
+      `Connected to Mongo! Database name: "${x.connections[0].name}"`
+    );
   })
   .catch(err => {
-    console.error('Error connecting to mongo', err)
+    console.error("Error connecting to mongo", err);
   });
 
-const app_name = require('./package.json').name;
-const debug = require('debug')(`${app_name}:${path.basename(__filename).split('.')[0]}`);
+const app_name = require("./package.json").name;
+const debug = require("debug")(
+  `${app_name}:${path.basename(__filename).split(".")[0]}`
+);
 
 const app = express();
 
 // Middleware Setup
-app.use(logger('dev'));
+app.use(logger("dev"));
 app.use(bodyParser.json());
 app.use(bodyParser.urlencoded({ extended: false }));
 app.use(cookieParser());
 
-// Express View engine setup
+app.use(
+  session({
+    secret: "keyboard cat",
+    resave: true,
+    saveUninitialized: true,
+    store: new MongoStore({
+      mongooseConnection: mongoose.connection
+    })
+  })
+);
 
-app.use(require('node-sass-middleware')({
-  src:  path.join(__dirname, 'public'),
-  dest: path.join(__dirname, 'public'),
-  sourceMap: true
-}));
-      
+// Express View engine setup
 
-app.set('views', path.join(__dirname, 'views'));
-app.set('view engine', 'hbs');
-app.use(express.static(path.join(__dirname, 'public')));
-app.use(favicon(path.join(__dirname, 'public', 'images', 'favicon.ico')));
+app.use(
+  require("node-sass-middleware")({
+    src: path.join(__dirname, "public"),
+    dest: path.join(__dirname, "public"),
+    sourceMap: true
+  })
+);
 
+app.set("views", path.join(__dirname, "views"));
+app.set("view engine", "hbs");
+hbs.registerPartials(__dirname + "/views/partials");
+app.use(express.static(path.join(__dirname, "public")));
+app.use(favicon(path.join(__dirname, "public", "images", "favicon.ico")));
 
+// Configura usuario como variable local
+app.use((req, res, next) => {
+  console.log(req.user);
+  res.locals.user = req.session.currentUser;
+  next();
+});
 
 // default value for title local
-app.locals.title = 'Express - Generated with IronGenerator';
+app.locals.title = "Express-Auth";
 
+const index = require("./routes/index");
+app.use("/", index);
 
+const auth = require("./routes/auth");
+app.use("/auth", auth);
 
-const index = require('./routes/index');
-app.use('/', index);
+const main = require("./routes/main");
```

Comment:
Como podéis comprobar la lógica del servidor se compone de muchos ficheros conectados entre sí, una buena práctica es ir dejando el fichero `app.js` lo más simple posible. Un paso sencillo para eso es importar sólo el fichero `index.js` de las rutas y en él importar el resto de `routers` configurados.
Os paso un ejemplo de cómo puede quedar el fichero `index.js`
```
const express = require("express");
const auth = require("./auth");
const private = require("./private");

const router = express.Router();

router.use("/auth", auth);
router.use("/private", private);

router.get("/", (req, res, next) => {
  if (req.session.currentUser)
    return res.render("private/home", { user: req.session.currentUser });

  res.redirect("/auth/login");
});

module.exports = router;
```